### DISCUSSION TITLE
Error: Your CLT does not support macOS 11.2.

### DISCUSSION THREAD

**Discussion by brunzefb** [Homebrew/discussions#913] [Everyday usage]
Hi,

after running
`brew upgrade`
I am getting
`Error: Your CLT does not support macOS 11.2.`

I have xcodebuild at version 12.4 (Build version 12D4e), the latest released version for BigSur 11.2.1

How can I get brew to work again?

**Comment by SMillerDev** [comment]
What does `brew config` and `brew doctor` say?

**Reply by brunzefb** [reply]
```
╰─ᐅ brew config
HOMEBREW_VERSION: 2.5.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: b43c0fed789b4cae33cd200284c44a095db57c3c
Last commit: 3 months ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 18e8999cf914d426313665679ca76dc464a593ac
Core tap last commit: 3 months ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.QTAPWte65T/org.macosforge.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 12.0 build 1200
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 10.0.2
macOS: 11.2.1-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
XQuartz: 2.7.11 => /opt/X11

╰─ᐅ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed with `brew diy`.
You should find replacements for the following formulae:
  python@2

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  helm@2
  ntfs-3g
  protobuf@3.7

Warning: Found Ruby file outside rhyeal/aws-rotate-iam-keys tap formula directory.
(/usr/local/Homebrew/Library/Taps/rhyeal/homebrew-aws-rotate-iam-keys/Formula):
  /usr/local/Homebrew/Library/Taps/rhyeal/homebrew-aws-rotate-iam-keys/aws-rotate-iam-keys.template.rb

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  coreutils
  docker
  python@2
  minikube
  python@3.9

Warning: You are using macOS 11.2.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or any other official channels. You are responsible for
resolving any issues you experience while you are running this
pre-release version.


Warning: Your CLT does not support macOS 11.2.
It is either outdated or was modified.
Please update your CLT or delete it if no updates are available.
```

**Reply by SMillerDev** [reply]
You should really really run `brew update` at the very least and preferably `brew upgrade` too. Your homebrew install is too old to recognize macOS 11.2

**Reply by carlocab** [reply]
Your version of `brew` is badly out of date. A `brew update` should fix your problems.

You also have `HOMEBREW_NO_AUTO_UPDATE` set. Either unset it, or remember to run `brew update` occasionally, to avoid finding yourself in this situation again in the future.

**Reply by brunzefb** [reply]
Hi -- the problem is that this error happens when I run 
`brew upgrade`
I had turned off the auto-update, because it is annoying -- it takes forever, and sometimes I just want to install a package without having to wait.

What's the manual procedure (git pull?) to update homebrew to avoid the error?

**Reply by carlocab** [reply]
I'd rather not recommend a manual update procedure. It's easy to figure out if you know what you're doing, but also easy to mess up if you don't.

As mentioned above, you need to run `brew update` right now. This will take some time, as you haven't done it in at least three months. The manual procedure won't be any faster: it's likely to be slower and, even if it isn't, it could lead you into some subtle and difficult-to-diagnose problems.

What you really need to do is 

>You also have `HOMEBREW_NO_AUTO_UPDATE` set. Either unset it, or remember to run `brew update` occasionally, to avoid finding yourself in this situation again in the future.

You can also adjust `HOMEBREW_AUTO_UPDATE_SECS` (default: `300`) to avoid updating each time you run `brew install` or `brew upgrade`. Alternatively, you can set up a `launchd` or `cron` job to run `brew update` regularly so that you don't have to.

**Reply by gromgit** [reply]
> What's the manual procedure (git pull?) to update homebrew to avoid the error?

As both @SMillerDev and @carlocab have mentioned, `brew update`. `HOMEBREW_NO_AUTO_UPDATE` by itself isn't a problem if you trigger regular updates. For instance, I set up a cron job to run `brew update` at 5am every day, so I don't have to wait at a terminal to keep my Homebrew setup up-to-date.

**Reply by Bo98** [reply]
`brew update` should not print that error. Usually, `brew upgrade` would also invoke `brew update` but your `HOMEBREW_NO_AUTO_UPDATE` prevents that.

`brew update` = update `brew`
`brew upgrade` = upgrades installed packages (and invoke `brew update` unless you have `HOMEBREW_NO_AUTO_UPDATE`).
